aesara/compile/profiling.py

```diff
@@ -37,93 +49,92 @@
-            destination_file = sys.stderr
+            destination_file = "<stderr>"
         elif config.profiling__destination == "stdout":
-            destination_file = sys.stdout
+            destination_file = "<stdout>"
         else:
-            destination_file = open(config.profiling__destination, "w")
-
-        # Reverse sort in the order of compile+exec time
-        for ps in sorted(
-            _atexit_print_list, key=lambda a: a.compile_time + a.fct_call_time
-        )[::-1]:
-            if (
-                ps.fct_callcount >= 1
-                or ps.compile_time > 1
-                or getattr(ps, "callcount", 0) > 1
-            ):
-                ps.summary(
+            destination_file = config.profiling__destination
+
+        with extended_open(destination_file, mode="w"):
+
+            # Reverse sort in the order of compile+exec time
+            for ps in sorted(
+                _atexit_print_list, key=lambda a: a.compile_time + a.fct_call_time
+            )[::-1]:
+                if (
+                    ps.fct_callcount >= 1
+                    or ps.compile_time > 1
+                    or getattr(ps, "callcount", 0) > 1
+                ):
+                    ps.summary(
+                        file=destination_file,
+                        n_ops_to_print=config.profiling__n_ops,
+                        n_apply_to_print=config.profiling__n_apply,
+                    )
+
+                    if ps.show_sum:
+                        to_sum.append(ps)
+                else:
+                    # TODO print the name if there is one!
+                    print("Skipping empty Profile")
+            if len(to_sum) > 1:
+                # Make a global profile
+                cum = copy.copy(to_sum[0])
+                msg = f"Sum of all({len(to_sum)}) printed profiles at exit."
+                cum.message = msg
+                for ps in to_sum[1:]:
+                    for attr in [
+                        "compile_time",
+                        "fct_call_time",
+                        "fct_callcount",
+                        "vm_call_time",
+                        "optimizer_time",
+                        "linker_time",
+                        "validate_time",
+                        "import_time",
+                        "linker_node_make_thunks",
+                    ]:
+                        setattr(cum, attr, getattr(cum, attr) + getattr(ps, attr))
+
+                    # merge dictionary
+                    for attr in [
+                        "apply_time",
+                        "apply_callcount",
+                        "apply_cimpl",
+                        "variable_shape",
+                        "variable_strides",
+                        "variable_offset",
+                        "linker_make_thunk_time",
+                    ]:
+                        cum_attr = getattr(cum, attr)
+                        for key, val in getattr(ps, attr.items()):
```

Comment:
@brandonwillard `attr` is a string, should this line be `for key, val in getattr(ps, attr).items():`

Reply:
Looks like it.  As you can tell, this is old code with apparently no coverage (or type checking), so we'll need to add some tests and type hints when making this correction.